sunset/code/modules/busy_space/air_traffic.dm

```diff
@@ -0,0 +1,129 @@
+//Cactus, Speedbird, Dynasty, oh my
+
+GLOBAL_DATUM_INIT(atc, /datum/lore/atc_controller, new)
+
+/datum/lore/atc_controller
+	var/delay_max = 10 MINUTES			//Maximum amount of tiem between ATC messages.  Default is 10 mins.
+	var/delay_min = 5 MINUTES			//Minimum amount of time between ATC messages.  Default is 5 mins.
+	var/backoff_delay = 5 MINUTES		//How long to back off if we can't talk and want to.  Default is 5 mins.
+	var/next_message					//When the next message should happen in world.time
+	var/force_chatter_type				//Force a specific type of messages
+
+	var/squelched = FALSE				//If ATC is squelched currently
+
+/datum/lore/atc_controller/New()
+	addtimer(CALLBACK(src, .proc/msg, "New shift beginning, resuming traffic control."), 30 SECONDS)
+	next_message = world.time + rand(delay_min, delay_max)
+	atc_process()
+
+/datum/lore/atc_controller/proc/atc_process()
+	if(world.time >= next_message)
+		if(squelched)
+			next_message = world.time + backoff_delay
+		else
+			next_message = world.time + rand(delay_min,delay_max)
+			random_convo()
+
+	addtimer(CALLBACK(src, .proc/atc_process), 1 MINUTES)
```

Comment:
Why the unholy mix of timers and world.time. Use timer delays instead.

Reply:
god no i don't want to refactor half of this